File: src/main/kotlin/com/zacksimpson/reminders/data/FirebaseConfig.kt
package com.zacksimpson.reminders.data

import com.zacksimpson.reminders.BuildConfig

/**
 * The reminders-web Firebase project this app syncs against.
 *
 * [API_KEY] comes from [BuildConfig.FIREBASE_API_KEY], set via the gitignored
 * `local.properties` — never hardcode it here. It's Android-app restricted (scoped to
 * [ANDROID_PACKAGE] + [ANDROID_CERT_SHA1]), a separate key from reminders-web's own web
 * key since a Google Cloud API key can't be both referrer- and Android-restricted.
 * A build signed with a different cert will need its SHA-1 added to the key's allowed
 * applications via `gcloud services api-keys update`.
 */
object FirebaseConfig {
    val API_KEY: String = BuildConfig.FIREBASE_API_KEY
    const val PROJECT_ID = "reminders-web-zs2026"

    // Sent as X-Android-Package/X-Android-Cert on every request — a raw REST call has
    // no Play-Services app identity, so these headers satisfy the restriction manually.
    const val ANDROID_PACKAGE = "com.zacksimpson.reminders"
    const val ANDROID_CERT_SHA1 = "B822D481499EE5C3CC7DB90AEEC2945D8F230715"

    // Sent as X-Firebase-gmpid — the app ID every Firebase SDK attaches to itself.
    const val FIREBASE_APP_ID = "1:1055613872869:web:5d1911c7d04e6819e5020c"
}

File: utils/sync/firebaseConfig.ts
/**
 * The reminders-web Firebase project this app syncs against.
 *
 * API_KEY comes from EXPO_PUBLIC_FIREBASE_API_KEY (a gitignored .env.local locally,
 * an EAS secret/env for cloud builds), never hardcode it here. It's Android-app
 * restricted in Google Cloud (scoped to a set of allowed (package, cert) pairs), which
 * is why ANDROID_PACKAGE/ANDROID_CERT_SHA1 below have to be sent on every request.
 */
export const FirebaseConfig = {
  API_KEY: process.env.EXPO_PUBLIC_FIREBASE_API_KEY ?? "",
  PROJECT_ID: "reminders-web-zs2026",

  // Sent as X-Android-Package/X-Android-Cert on every request: a raw REST call has no
  // Play-Services app identity, so these headers satisfy the restriction manually.
  // Dev builds (bun run dev) get a ".debug" applicationId suffix at the Gradle level
  // (see plugins/withDebugApplicationIdSuffix.js); EAS preview/production builds
  // don't, so this mirrors the real package name for each.
  ANDROID_PACKAGE:
    process.env.NODE_ENV === "production"
      ? "com.im360noscope.reminderstool"
      : "com.im360noscope.reminderstool.debug",
  // Must match whichever keystore actually signs the running build: the local debug
  // keystore for `bun run dev`, or the EAS-managed keystore for preview/production.
  // `keytool`/`gradlew signingReport` print this colon-separated (e.g. "AB:CD:..."),
  // but the X-Android-Cert header Google checks against needs raw hex with no
  // separators, so strip them here to let .env.local hold either format.
  ANDROID_CERT_SHA1: (
    process.env.EXPO_PUBLIC_FIREBASE_ANDROID_CERT_SHA1 ?? ""
  )
    .replace(/:/g, "")
    .toUpperCase(),

  // Sent as X-Firebase-gmpid: the app ID every Firebase SDK attaches to itself.
  // Same project-wide web app id the native rewrite also reuses, not restricted per
  // package, so safe to hardcode.
  FIREBASE_APP_ID: "1:1055613872869:web:5d1911c7d04e6819e5020c",
};
